Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1029 @@
+      old.author !== current.author ||
+      old.copyright !== current.copyright ||
+      old.link !== current.link ||
+      old.time !== current.time ||
+      old.bounds !== current.bounds
+    ) {
+      return true;
+    }
+    return false;
+  }
+  return true;
+}
+
+function load(dataSource, entityCollection, data, options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  var sourceUri = options.sourceUri;
+
+  var promise = data;
+  if (typeof data === "string" || data instanceof Resource) {
+    data = Resource.createIfNeeded(data);
+    promise = data.fetchBlob();
+    sourceUri = defaultValue(sourceUri, data.clone());
+
+    // Add resource credits to our list of credits to display
+    var resourceCredits = dataSource._resourceCredits;
+    var credits = data.credits;
+    if (defined(credits)) {
+      var length = credits.length;
+      for (var i = 0; i < length; i++) {
+        resourceCredits.push(credits[i]);
+      }
+    }
+  } else {
+    sourceUri = defaultValue(sourceUri, Resource.DEFAULT.clone());
+  }
+
+  sourceUri = Resource.createIfNeeded(sourceUri);
+
+  return when(promise)
+    .then(function (dataToLoad) {
+      if (dataToLoad instanceof Blob) {
+        return readBlobAsText(dataToLoad).then(function (text) {
+          //There's no official way to validate if a parse was successful.
+          //The following check detects the error on various browsers.
+          //IE raises an exception
+          var gpx;
+          var error;
+          try {
+            gpx = parser.parseFromString(text, "application/xml");
+          } catch (e) {
+            error = e.toString();
+          }
+
+          //The parse succeeds on Chrome and Firefox, but the error
+          //handling is different in each.
+          if (
+            defined(error) ||
+            gpx.body ||
+            gpx.documentElement.tagName === "parsererror"
+          ) {
+            //Firefox has error information as the firstChild nodeValue.
+            var msg = defined(error)
+              ? error
+              : gpx.documentElement.firstChild.nodeValue;
+
+            //Chrome has it in the body text.
+            if (!msg) {
+              msg = gpx.body.innerText;
+            }
+
+            //Return the error
+            throw new RuntimeError(msg);
+          }
+          return loadGpx(dataSource, gpx, sourceUri, undefined);
+        });
+      }
+    })
+    .otherwise(function (error) {
+      dataSource._error.raiseEvent(dataSource, error);
+      console.log(error);
+      return when.reject(error);
+    });
+}
+
+/**
+ * A {@link DataSource} which processes The GPS Exchange Format (GPX).
+ *
+ * @alias GpxDataSource
+ * @constructor
+ *
+ * @param {DefaultProxy} [proxy] A proxy to be used for loading external data.
+ *
+ * @see {@link http://www.topografix.com/gpx.asp|Topografix GPX Standard}
+ * @see {@link http://www.topografix.com/gpx/1/1/|Topografix GPX Documentation}
+ *
+ * @demo {@link}
```

Comment:
I think this should link to the Sandcastle? You can see what this looks like by running `npm run generateDocumentation` and going to http://localhost:8080/Build/Documentation/GpxDataSource.html?classFilter=GPX. I would see what other classes do.